### ISSUE TITLE
CLI/Watch - Extract with [files] marks other strings as obsolete in messages file

### ISSUE THREAD

**Issue by jack1913** [lingui/js-lingui#966]
**Describe the bug**
I am in the middle of writing a little script to automate extraction of strings using `chockidar` as watcher.

Unfortunately I have came across 2 errors.

1. **CLI** - Whenever I pass a `[files]` string into the cmd it is not recognized and command is extracting from all source path provided in config file of lingui. But for my case it is not really important, just pointing this out. Ex. command `lingui extract src/pages/SignIn/SignIn.tsx --verbose`.
2. **Main problem** - after hitting the default function of `@lingui/cli/lingui-extract` with a `linguiConfig` and `linguiOptions` prepared based on the source code I have managed to get the "cli" to extract from the file I pin-pointed, BUT after the extraction all other previously extracted strings from other files are marked as obsolete.

**To Reproduce**
Just run this code via npm. It is just a base code to run the extraction from script.

```
const { getConfig, LinguiConfig } = require('@lingui/conf');

const linguiExtract = require('../node_modules/@lingui/cli/lingui-extract').default;

const linguiConfig = getConfig({
  configPath: process.env.LINGUI_CONFIG,
});

const linguiOptions = {
  verbose: true,
  clean: false,
  overwrite: true,
  files: ['src/pages/SignIn/SignIn.tsx'],
};

function prepareLinguiExtract(config, options) {
  linguiExtract(config, options);
}

prepareLinguiExtract(linguiConfig, linguiOptions);
```

**Expected behavior**
I expect it to just extract string from the file I am pointing to, not to modify the whole messages file.

**Additional context**
- jsLingui version 3.4.0
- Babel version 7.12.3
- CRA ejected, React 17.0.1

**Comment by jack1913** [comment]
I have just took a quick look through the code of `@lingui/cli` and found what lines could/should be changed to make this work as intended (as I assume).

https://github.com/lingui/js-lingui/blob/14f24f2725771dcea0793de146abc9b86ea88789/packages/cli/src/api/catalog.ts#L233-L239

These lines do not really get into account the fact of providing `[files]` option. I think if you could make it like this it would work as intended.

```
    // Mark all remaining translations as obsolete
    const obsoleteMessages = obsoleteKeys.map((key) => ({
      [key]: {
        ...prevCatalog[key],
        **obsolete: options.files ? false : true,**
      },
    }))
```

That is the SIMPLE version. Maybe it would also be possible to write a version of the code to mark all obsolete messaged from the specified files, without marking as obsolete all other translations but I did not (for now) look through the whole code.

**Comment by semoal** [comment]
Hm Jack youre probably right. tomorrow we have a planned release, could you create a PR fixing this and we'll release it tomorrow?

**Comment by jack1913** [comment]
@semoal I have made the pull request - https://github.com/lingui/js-lingui/pull/967

**Comment by semoal** [comment]
Jack, I'm thinking calmly about your idea and what about to integrate this to Lingui in a native way?

```sh
lingui extract --watch
```

This will run the chokidar instance (or any alternative of file-watcher) and run extract only in the changed files. For me looks a pretty interesting addition to Lingui.

Do you want to work on it?

**Comment by jack1913** [comment]
Sure, I can work on this. I already have a working code I can implement it natively, but I will sit to it by the end of the week.

**Comment by jack1913** [comment]
@semoal I have commited the v1 of the watch mode, pls check out if this implementation is acceptable.

I've added 2 functionalities:
1. `--watch` for `lingui extract`
2. `--watch` for `lingui compile` - I think in some implementations of Lingui it can be useful.

**Comment by semoal** [comment]
Released your funcitonality with 3.6.0. Thanks for the contribution!
